app/jobs/renewal_job.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class RenewalJob < ApplicationJob
+  queue_as :default
+
+  RENEWAL_CUSTOM_MESSAGELIST = {
+    "hatErrorResponse.141": 'Renewal limit reached, cannot be renewed.',
+    "hatErrorResponse.7703": 'Renewal limit reached, cannot be renewed.',
+    "hatErrorResponse.105": 'Item has been recalled, cannot be renewed.',
+    "hatErrorResponse.252": 'Item has holds, cannot be renewed.',
+    "hatErrorResponse.46": 'Item on reserve, cannot be renewed.',
+    "unhandledException": ''
```

Comment:
For this final case, when does it show up again? I am seeing it when trying to renew items that are overdue. Are there other cases?

Full json:

```
=> {"messageList"=>[{"code"=>"unhandledException", "message"=>"Unhandled exception: class java.lang.NullPointerException"}]}
```

Reply:
We were not showing any specific error msg for these case, was only showing the generic top msg when we were doing the flashes. But will be handled differently with Toast for sure and yeah probably good idea to add a msg here already. Thanks, I will add one.

It is not only overdue necessarily, can be other weird cases. This was one of the sirsi webservices bug. And yes all the overdue ones hit this bug.

Reply:
We can deal with this problem in the toasts issue 👍